src/Symfony/Component/EventDispatcher/Tests/EventDispatcherTest.php

```diff
@@ -412,6 +412,33 @@ public function testMutatingWhilePropagationIsStopped()
 
         $this->assertTrue($testLoaded);
     }
+
+    /**
+     * @group legacy
+     * @expectedDeprecation Calling the "Symfony\Component\EventDispatcher\EventDispatcherInterface::dispatch()" method with the event name as the first argument is deprecated since Symfony 4.3, pass it as the second argument and provide the event object as the first argument instead.
+     */
+    public function testLegacySignatureWithoutEvent()
+    {
+        $this->dispatcher->dispatch('foo');
+    }
+
+    /**
+     * @group legacy
+     * @expectedDeprecation Calling the "Symfony\Component\EventDispatcher\EventDispatcherInterface::dispatch()" method with the event name as the first argument is deprecated since Symfony 4.3, pass it as the second argument and provide the event object as the first argument instead.
+     */
+    public function testLegacySignatureWithEvent()
+    {
+        $this->dispatcher->dispatch('foo', new Event());
+    }
+
+    /**
+     * @expectedException \TypeError
+     * @expectedExceptionMessage Argument 1 passed to "Symfony\Component\EventDispatcher\EventDispatcherInterface::dispatch()" must be an instance of Symfony\Component\EventDispatcher\Event, string given.
```

Comment:
That doesn't seem right to me. Shouldn't it say argument 1 must be of type object? That is what would happen when we add the type hints in 5.0 and that is what is documented in phpdoc.

Reply:
good catch, fixed